pkg/kubeapiserver/options/authentication.go

```diff
@@ -574,7 +575,7 @@ func (o *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 		}
 	}
 
-	if err := apiservervalidation.ValidateAuthenticationConfiguration(ret.AuthenticationConfig, ret.ServiceAccountIssuers).ToAggregate(); err != nil {
+	if err := apiservervalidation.ValidateAuthenticationConfiguration(authenticationcel.NewDefaultCompiler(), ret.AuthenticationConfig, ret.ServiceAccountIssuers).ToAggregate(); err != nil {
```

Comment:
it's not clear to me what principle is used to decide whether to plumb in a configurable compiler (like staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go does) and what chooses to call the global default constructor, like this does ... how did you decide which to do in each call site?

Reply:
Validation is stateless and hence gets the argument, always. Here I am considering the compiler with its heavy initialization as state, i.e. it belongs onto the heap, not the stack.

Constructors like for oidc or webhooks return state, so they are in general fine with construction of another compiler. But I have made it shareable/configurable for those constructors that showed up in pprof.